### ISSUE TITLE
Blueprint sync plugin

### ISSUE THREAD

**Issue by Hooolm** [clusterio/clusterio#476]
I found myself instantly creating a blueprint world. To keep that entirely separate. Nothing synced only jump to and from to do designs in there.

However, since I do not want item sync, nor research sync. **It would be awesome to have blueprint sync somehow.** I now how to export and import across all instances to update the game blueprints for me and my friends that I play with.

**Comment by psihius** [comment]
> I found myself instantly creating a blueprint world. To keep that entirely separate. Nothing synced only jump to and from to do designs in there.
> 
> However, since I do not want item sync, nor research sync. **It would be awesome to have blueprint sync somehow.** I now how to export and import across all instances to update the game blueprints for me and my friends that I play with.

That's what the Blueprint Library of the game is for - stick your blueprints in there and they will follow you around.

P.S. If you put a blueprint in your inventory, it will be synced as an item.

**Comment by PsychoZander** [comment]
I do see where he is coming from though. It's for shared servers, where you want a single repository of blueprints accessible to all players on the server. Having to transfer them manually from one server to another is of course possible by either using Item Sync or the personal blueprint library, however it would be more graceful to have a plugin that handles that for you.

**Comment by Hooolm** [comment]
https://user-images.githubusercontent.com/1484634/196134023-e05c6850-89d4-424b-a039-f5473c7a56bc.mp4

In this video i expected my Game blueprints to be shared, meaning that the "Ping?" book would appear after i jump to my play server. However, there are no syncs setup between these servers. No items, nor research. Only used as creative world.

**Comment by feupeu** [comment]
Me and my friends have a similar issue on our instances. We're currently limited by:

- Manually having to import/export blueprints (we can't use discord messages for this, since the strings are too large, so we use a git repository)
- We can only use a single book. People would use old versions of blueprints if they cannot simple import a single blueprint string (having to export/import multiple blueprints would be too much work, but having a single book is not great for player experience, since each time we want to stamp down a blueprint, we have to go through an extra layer that we don't really want)
- When a new update to the blueprint is introducted (in git) we have to go through each instance to update it in the game blueprints (people are tempted to put the blueprint book into their inventory, since they would otherwise have to import it on all instances, which would lead to them using old versions of the book)

TL;DR. Having multiple instances without synchronized "game blueprints" is a pain for a multiplayer playthrough. Especially in our scenario, where we're heavily modded and focused around versioned blueprints. Having game blueprints synchronized across instances would be a great feature for us.

I do however see how migration of existing saves could be troublesome, and having instances with different mods could lead to some blueprints being invalid (they might use items from a mod that is not installed)

**Comment by Hornwitser** [comment]
@Hooolm @feupeu there is no API to interact with the Game Blueprint library so it's simply not possible to implement the feature you describe with the current state of the game.

**Comment by Hornwitser** [comment]
I've posted a Mod Interface request about it at https://forums.factorio.com/103772.
